### ISSUE TITLE
Add option to allow focusing editor (file content) on single click in sidebar

### ISSUE THREAD

**Issue by porsager** [zed-industries/zed#8652]
### Check for existing issues

- [X] Completed

### Describe the feature

Multiple times daily I accidentally start typing after selecting a file in the sidebar because I expect the text editor to be focused.

I found out I can double click the files, but that feels really unintuitive when they already open up by a single click.

It is probably to late to change the default, but having an option to switch would really improve the experience and follow what other editors are doing.

**Comment by Moshyfawn** [comment]
Related #4374

**Comment by maxbrunsfeld** [comment]
I agree that we should not change the default behavior; our current behavior matches VS code's, and is probably familiar to a lot of people. But I'm open to adding a setting for this if lots of people have this expectation.
